### PR TITLE
Codemeta harvesting

### PR DESCRIPTION
**Pull request by sdruskat**
This PR implements CodeMeta harvesting at its simplest. Implementation is POC still, I'd like to get some validation in there if possible. Still, comments are welcome. I left TODOs in there as reminders.

### CODE REVIEW

File: src/hermes/commands/harvest/codemeta.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    Implementation of a harvester that provides data from a codemeta.json file format.
+
+    :param click_ctx: Click context that this command was run inside (might be used to extract command line arguments).
+    :param ctx: The harvesting context that should contain the provided metadata.
+    """
+    # Get the parent context (every subcommand has its own context with the main click context as parent)
+    parent_ctx = click_ctx.parent
+    if parent_ctx is None:
+        raise RuntimeError('No parent context!')
+    path = parent_ctx.params['path']
```
Comment (led02) [inline src/hermes/commands/harvest/codemeta.py:23]: This is kind of repeating...

File: test/hermes_test/commands/harvest/test_codemeta.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+def codemeta():
+    return json.loads(CODEMETA_JSON)
+
+
+@pytest.fixture()
+def valid_codemeta(tmp_path):
+    codemeta_json = json.loads(CODEMETA_JSON)
+    codemeta_file = tmp_path / 'codemeta.json'
+    json.dump(codemeta_json, codemeta_file)
+    return codemeta_file
```
Comment (led02) [inline test/hermes_test/commands/harvest/test_codemeta.py:130]: Is this fixture used at all?

File: src/hermes/commands/harvest/codemeta.py (hunk 1 above)
Comment (sdruskat) [reply src/hermes/commands/harvest/codemeta.py:23]: @led02 You mean across harvesters (same code is in the CFF harvester). I can refactor this into either `__init__` or create a superclass for harvesters in general...

**Comment by led02** [comment]
Included in #40

**Comment by jkelling** [comment]
> Included in #40

Shall close this one then?

**Review by led02 — APPROVED** [review]

**Pull request merged**